crates/bitwarden/src/auth/login/access_token.rs

```diff
@@ -82,12 +110,43 @@ async fn request_access_token(
         .await
 }
 
+fn load_tokens_from_state(
+    client: &mut Client,
+    state_file: &Path,
+    access_token: &AccessToken,
+) -> Result<Uuid> {
+    let client_state = state::get(state_file, access_token)?;
+
+    if let Some(client_state) = client_state {
+        let token: JWTToken = client_state.token.parse()?;
+
+        if let Some(organization_id) = token.organization {
+            let time_till_expiration = (token.exp as i64) - Utc::now().timestamp();
+
+            if time_till_expiration > 0 {
+                let organization_id: Uuid = organization_id
+                    .parse()
+                    .map_err(|_| Error::Internal("Bad organization id."))?;
+                let encryption_key: SymmetricCryptoKey = client_state.encryption_key.parse()?;
+
+                client.set_tokens(client_state.token, None, time_till_expiration as u64);
+                client.initialize_crypto_single_key(encryption_key);
+
+                return Ok(organization_id);
+            }
+        }
+    }
+
+    Err(Error::Internal("Could not load tokens from state."))
+}
+
 /// Login to Bitwarden with access token
 #[derive(Serialize, Deserialize, Debug, JsonSchema)]
 #[serde(rename_all = "camelCase", deny_unknown_fields)]
 pub struct AccessTokenLoginRequest {
     /// Bitwarden service API access token
     pub access_token: String,
+    pub state_file: Option<PathBuf>,
```

Comment:
This is a breaking change, let's update the changelog, and ensure we bump to 0.4.

Reply:
Resolved in [0c8125f](https://github.com/bitwarden/sdk/pull/388/commits/0c8125f5f38d78610de24b249384219cdc83fc07)